Source/diablo.cpp

```diff
@@ -67,6 +67,8 @@ WORD tick_delay = 50;
  * screen, as needed for efficient rendering in fullscreen mode.
  */
 BOOL fullscreen = TRUE;
+BOOL hwcursor = TRUE;
+int windowResVer = 1;

```

Comment:
Something like windowsResolutionVersion seemed a bit too bulky to type and read. Basically this variable holds version of the game windo resolution. It's incremented with every size or fullscreen mode change so that we know the cursor size need to be addressed as well.

Reply:
The value 1 is there to differentiate the initial value from other global variables that might be compared with windowResVer, so that 'something != windowResVer' holds true initially.

Reply:
Again, apologies for the potentially noob question, but what does a "version" of a resolution even mean? I just fail to grasp the concept here. I don't understand why we need to keep track of an incrementing value on resolution change. Doesn't the cursor only depend on the current resolution?

Reply:
It's because the implementation isn't event based and instead compares state on every frame.

It would be good to change this to be event based since SDL emits events for these changes that we can then process as they happen.